apps/channels/forms.py

```diff
@@ -532,3 +533,92 @@ class CommCareConnectChannelForm(ExtraFormBase):
         max_length=100,
     )
+
+
+class EmbeddedWidgetChannelForm(ExtraFormBase):
+    """Form for creating embedded chat widget channels"""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # Initialize widget data attributes
+        self.widget_token = None
+        self.embed_code = None
+
+        # Convert allowed_domains list back to newline-separated text for editing
+        if self.channel and self.channel.extra_data.get("allowed_domains"):
+            domains_list = self.channel.extra_data.get("allowed_domains", [])
+            if isinstance(domains_list, list) and domains_list:
+                self.fields["allowed_domains"].initial = "\n".join(domains_list)
+
+    allowed_domains = forms.CharField(
+        label="Allowed Domains",
+        widget=forms.Textarea(
+            attrs={
+                "rows": 4,
+                "class": "textarea textarea-bordered w-full",
+                "placeholder": "Enter one domain per line, e.g.:\nexample.com\nwww.mysite.org\nsubdomain.example.com",
+            }
+        ),
+        help_text="Enter the domains where this widget is allowed to be embedded (one per line). "
+        "Leave blank to allow all domains.",
+        required=False,
+    )
+
+    def clean_allowed_domains(self):
+        """Validate and clean the allowed domains"""
+        domains_text = self.cleaned_data.get("allowed_domains", "").strip()
+        if not domains_text:
+            return []
+
+        domains = []
+        for line in domains_text.split("\n"):
+            domain = line.strip()
+            if domain:
+                if not re.match(
+                    r"^(\*\.)?[a-zA-Z0-9]([a-zA-Z0-9-]*[a-zA-Z0-9])?(\.[a-zA-Z0-9]([a-zA-Z0-9-]*[a-zA-Z0-9])?)*(:\d+)?$",
+                    domain,
+                ):
+                    raise forms.ValidationError(f"Invalid domain format: {domain}")
+                domains.append(domain)
+        return domains
+
+    def clean(self):
+        """Generate or preserve the widget token"""
+        cleaned_data = super().clean()
+
+        # If editing existing channel, preserve the token
+        if self.channel and self.channel.extra_data.get("widget_token"):
+            cleaned_data["widget_token"] = self.channel.extra_data["widget_token"]
+        else:
+            # Generate token here so it's available when check_usage_by_another_experiment is called
+            cleaned_data["widget_token"] = secrets.token_urlsafe(24)
+
+        return cleaned_data
+
+    def post_save(self, channel: ExperimentChannel):
+        """Save widget data and set success message"""
+        widget_token = self.cleaned_data["widget_token"]
+        embed_code = self._generate_embed_code(channel, widget_token)
+        channel.extra_data.update(
+            {
+                "widget_token": widget_token,
+                "allowed_domains": self.cleaned_data["allowed_domains"],
+            }
+        )
+        channel.save()
```

Comment:
this is already handled in the channel form: https://github.com/dimagi/open-chat-studio/blob/33f64319ceb4759d63146f490c34276dfbfd01d1/apps/channels/forms.py#L69-L72